### PR TITLE
Preliminary packaging

### PR DESCRIPTION
**Pull request by lucasmccabe**
In this PR:
1. Minor organizational refactor so that major structs and functions are exported as part of one `Juissie` module for easier navigation.
2. Juissie is organized as a package with more coherent dependency management and installation.

To install the package dependencies:
```julia
using Pkg
Pkg.activate(".")
Pkg.instantiate()
```
This allows you to import the `Juissie` module:
```julia
using Juissie
```
and access its associated symbols, like `Corpus`, `Embedder`, `upsert_chunk`, `upsert_document`, `search`, and `embed`.

More detailed setup instructions are found in `guides/setup.md`.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name = "Juissie"
+uuid = "d87d79c4-7fb2-4087-a91e-b4c4d91349d0"
+authors = ["Lucas H. McCabe <lucasmccabe@gwu.edu>", "Arthur Bacon <arthurbacon@nocabsoftware.com>", "Alexey Iakovenko <alexey@iakovenko.com>", "Artin Yousefi <arthurbacon@nocabsoftware.com>"]
```
Comment (toon-leader-bacon) [inline Project.toml:3]:
```suggestion
authors = ["Lucas H. McCabe <lucasmccabe@gwu.edu>", "Arthur Bacon <arthurbacon@nocabsoftware.com>", "Alexey Iakovenko <alexey@iakovenko.com>", "Artin Yousefi <artinyousefi@gwmail.gwu.edu>"]
```

File: src/Juissie.jl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module Juissie
+
+include("SemanticSearch/SemanticSearch.jl")
+
+using .SemanticSearch
+export Corpus, upsert_chunk, upsert_document, search,
+    Embedder, embed
```
Comment (toon-leader-bacon) [inline src/Juissie.jl:7]: I guess this is the best way to encapsulate "classes" in Julia. Looks a little weird to my eyes, but it's just a style thing I guess.

**Review by toon-leader-bacon — COMMENTED** [review]
Looks good. One minor suggested change in the Project.toml but otherwise good. Will approve once the suggestion is addressed.

File: Project.toml (hunk 1 above)
Comment (lucasmccabe) [reply Project.toml:3]: Oops - good catch

**Pull request merged**